nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -0,0 +1,232 @@
+        "        + \"the object so that it can be fetched in conjunction with FetchS3Object. This Processor is designed to run on Primary Node only \"\n" +
+        "        + \"in a cluster. If the primary node changes, the new Primary Node will pick up where the previous node left off without duplicating \"\n" +
+        "        + \"all of the data.")
+@Stateful(scopes = Scope.CLUSTER, description = "After performing a listing of keys, the timestamp of the newest key is stored, "
+        + "along with the keys that share that same timestamp. This allows the Processor to list only keys that have been added or modified after "
+        + "this date the next time that the Processor is run. State is stored across the cluster so that this Processor can be run on Primary Node only and if a new Primary "
+        + "Node is selected, the new node can pick up where the previous node left off, without duplicating the data.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "s3.bucket", description = "The name of the S3 bucket"),
+        @WritesAttribute(attribute = "filename", description = "The name of the file"),
+        @WritesAttribute(attribute = "s3.etag", description = "The ETag that can be used to see if the file has changed"),
+        @WritesAttribute(attribute = "s3.lastModified", description = "The last modified time in milliseconds since epoch in UTC time"),
+        @WritesAttribute(attribute = "s3.length", description = "The size of the object in bytes"),
+        @WritesAttribute(attribute = "s3.storeClass", description = "The storage class of the object"),})
+@SeeAlso({FetchS3Object.class, PutS3Object.class, DeleteS3Object.class})
+public class ListS3 extends AbstractS3Processor {
+
+    public static final PropertyDescriptor DELIMITER = new PropertyDescriptor.Builder()
+            .name("delimiter")
+            .displayName("Delimiter")
+            .expressionLanguageSupported(false)
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("The string used to delimit directories within the bucket. Please consult the AWS documentation " +
+                    "for the correct use of this field.")
+            .build();
+
+    public static final PropertyDescriptor PREFIX = new PropertyDescriptor.Builder()
+            .name("prefix")
+            .displayName("Prefix")
+            .expressionLanguageSupported(true)
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("The prefix used to filter the object list. In most cases, it should end with a forward slash ('/').")
+            .build();
+
+    public static final List<PropertyDescriptor> properties = Collections.unmodifiableList(
+            Arrays.asList(BUCKET, KEY, REGION, ACCESS_KEY, SECRET_KEY, CREDENTIALS_FILE,
```

Comment:
Doesn't seem like KEY is being used, if that is the case, should be removed from properties.